### PR TITLE
[RNmobile] Update path to extension availability util

### PR DESCRIPTION
**Pull request by hypest**
Fixes the broken [native mobile build](https://app.circleci.com/pipelines/github/wordpress-mobile/gutenberg-mobile/16140/workflows/a6b3a6be-51de-4f77-bcb3-31e8a86e8b24/jobs/87670) by updating the paths to some util functions, moved recently via https://github.com/Automattic/jetpack/pull/22642.

#### Changes proposed in this Pull Request:

* Update path to `get-jetpack-extension-availability`

#### Jetpack product discussion

TBD

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Go to '..'
*

### CODE REVIEW

**Comment by hypest** [comment]
👋 @SiobhyB ! I'm keeping the PR still in draft mode as I'm not yet sure what are all the steps I need to complete, but added you as a reviewer as you probably already have experience here.

> #### Jetpack product discussion
> <\!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
> <\!-- Make sure any changes to existing products have been discussed and agreed upon -->

About the prompt for a product discussion, do you think there needs to be one related to this type of change?

**Comment by SiobhyB** [comment]
> 👋 @SiobhyB ! I'm keeping the PR still in draft mode as I'm not yet sure what are all the steps I need to complete, but added you as a reviewer as you probably already have experience here.

Thanks for the heads up! Ready to review when it's ready 🧐 

> About the prompt for a product discussion, do you think there needs to be one related to this type of change?

I don't feel it's really necessary for a change like this. Possibly linking to the Slack thread where you noted you'd be working on this (p1644399122157289-slack-C6UJ0KRKQ) could be useful in that section, for context? But I think skipping that section is also valid.

File: projects/plugins/jetpack/extensions/shared/register-jetpack-block.native.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { registerBlockType } from '@wordpress/blocks';
  * Internal dependencies
  */
 import extensionList from '../index.json';
-import getJetpackExtensionAvailability from './get-jetpack-extension-availability';
+import getJetpackExtensionAvailability from '../../../../js-packages/shared-extension-utils/src/get-jetpack-extension-availability';
```
Comment (jhnstn) [inline projects/plugins/jetpack/extensions/shared/register-jetpack-block.native.js:10]: Did you try `@automattic/get-jetpack-extension-availability` for the import statement? I wouldn't expect having to explicitly move up four directories to import that file.

**Comment by jeherve** [comment]
Related thread: pbArwn-3Ij-p2

File: projects/plugins/jetpack/extensions/shared/register-jetpack-block.native.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { registerBlockType } from '@wordpress/blocks';
  * Internal dependencies
  */
 import extensionList from '../index.json';
-import getJetpackExtensionAvailability from './get-jetpack-extension-availability';
+import { getJetpackExtensionAvailability } from '@automattic/jetpack-shared-extension-utils';
```
[the remark this answers is not in the document]
Comment (jhnstn) [reply projects/plugins/jetpack/extensions/shared/register-jetpack-block.native.js]: Thanks for that catch! resolved inhttps://github.com/Automattic/jetpack/pull/22763/commits/fde3d0911792049361cf61bea0f8ffb790aaf8a7

**Review by SiobhyB — APPROVED** [review]
Tested and confirmed that the changes in this PR resolve the Jetpack-related errors we were seeing in the app. I added a couple of Jetpack blocks (the Contact Form and Story blocks) with these branches checked out also, and confirm all worked well.

In addition, I checked out the changes on my WordPress.com sandbox to verify that there were no obvious regressions when adding blocks on the web. Approved! 🚀 

_Note: The `Tests / PHP tests: PHP 8.1 WP master (pull_request)` test is failing, but I understand that is optional (see `experimental` keyword [here](https://github.com/Automattic/jetpack/blob/74cb231cc423013ab9e947a65fda7aefe01e472f/.github/files/generate-ci-matrix.php#L66-L71) and [here](https://github.com/Automattic/jetpack/blob/74cb231cc423013ab9e947a65fda7aefe01e472f/.github/files/generate-ci-matrix.php#L41-L42).) I believe the E2E test is likely failing due to flakiness._

**Review by jeherve — APPROVED** [review]

**Comment by jeherve** [comment]
r240027-wpcom

**Pull request merged**